pkg/operator/ceph/object/realm/controller.go

```diff
@@ -215,6 +308,11 @@ func validateRealmCR(u *cephv1.CephObjectRealm) error {
 	if u.Namespace == "" {
 		return errors.New("missing namespace")
 	}
+	if u.Spec.IsPullRealm() {
```

Comment:
Seems like this check isn't needed. The error condition can never happen since `IsPullReam()` checks exactly the opposite of line 312, right?